diff --git a/filter.h b/filter.h
index 699ff32..89d5c0c 100644
--- a/filter.h
+++ b/filter.h
@@ -33,11 +33,11 @@
 /////////////////////////////////////////////////////////////////////////////
 
 __inline void hybrid_filter_dec(TTA_fltst *fs, TTAint32 *in) {
-	register TTAint32 *pA = fs->dl;
-	register TTAint32 *pB = fs->qm;
-	register TTAint32 *pM = fs->dx;
-	register TTAint32 sum = fs->round;
-	register __m128i xmA1, xmA2, xmB1, xmB2, xmM1, xmM2, xmDP;
+	TTAint32 *pA = fs->dl;
+	TTAint32 *pB = fs->qm;
+	TTAint32 *pM = fs->dx;
+	TTAint32 sum = fs->round;
+	__m128i xmA1, xmA2, xmB1, xmB2, xmM1, xmM2, xmDP;
 
 	xmA1 = _mm_load_si128((__m128i*)pA);
 	xmA2 = _mm_load_si128((__m128i*)(pA + 4));
@@ -83,11 +83,11 @@ __inline void hybrid_filter_dec(TTA_fltst *fs, TTAint32 *in) {
 /////////////////////////////////////////////////////////////////////////////
 
 __inline void hybrid_filter_enc(TTA_fltst *fs, TTAint32 *in) {
-	register TTAint32 *pA = fs->dl;
-	register TTAint32 *pB = fs->qm;
-	register TTAint32 *pM = fs->dx;
-	register TTAint32 sum = fs->round;
-	register __m128i xmA1, xmA2, xmB1, xmB2, xmM1, xmM2, xmDP;
+	TTAint32 *pA = fs->dl;
+	TTAint32 *pB = fs->qm;
+	TTAint32 *pM = fs->dx;
+	TTAint32 sum = fs->round;
+	__m128i xmA1, xmA2, xmB1, xmB2, xmM1, xmM2, xmDP;
 
 	xmA1 = _mm_load_si128((__m128i*)pA);
 	xmA2 = _mm_load_si128((__m128i*)(pA + 4));
@@ -135,10 +135,10 @@ __inline void hybrid_filter_enc(TTA_fltst *fs, TTAint32 *in) {
 /////////////////////////////////////////////////////////////////////////////
 
 __inline void hybrid_filter_dec(TTA_fltst *fs, TTAint32 *in) {
-	register TTAint32 *pA = fs->dl;
-	register TTAint32 *pB = fs->qm;
-	register TTAint32 *pM = fs->dx;
-	register TTAint32 sum = fs->round;
+	TTAint32 *pA = fs->dl;
+	TTAint32 *pB = fs->qm;
+	TTAint32 *pM = fs->dx;
+	TTAint32 sum = fs->round;
 
 	if (fs->error < 0) {
 		pB[0] -= pM[0]; pB[1] -= pM[1]; pB[2] -= pM[2]; pB[3] -= pM[3];
@@ -171,10 +171,10 @@ __inline void hybrid_filter_dec(TTA_fltst *fs, TTAint32 *in) {
 /////////////////////////////////////////////////////////////////////////////
 
 __inline void hybrid_filter_enc(TTA_fltst *fs, TTAint32 *in) {
-	register TTAint32 *pA = fs->dl;
-	register TTAint32 *pB = fs->qm;
-	register TTAint32 *pM = fs->dx;
-	register TTAint32 sum = fs->round;
+	TTAint32 *pA = fs->dl;
+	TTAint32 *pB = fs->qm;
+	TTAint32 *pM = fs->dx;
+	TTAint32 sum = fs->round;
 
 	if (fs->error < 0) {
 		pB[0] -= pM[0]; pB[1] -= pM[1]; pB[2] -= pM[2]; pB[3] -= pM[3];
diff --git a/libtta.h b/libtta.h
index b1b267c..a45bf91 100644
--- a/libtta.h
+++ b/libtta.h
@@ -18,7 +18,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h> 
+#if !defined(__APPLE__)
 #include <malloc.h> 
+#endif
 #include <stdexcept>
 
 #ifdef CARIBBEAN
